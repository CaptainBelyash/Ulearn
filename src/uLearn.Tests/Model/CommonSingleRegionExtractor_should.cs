using System.IO;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using uLearn.Model.Blocks;

namespace uLearn.Model
{
	[TestFixture]
	public class CommonSingleRegionExtractor_should
	{
		private static readonly DirectoryInfo dir = new DirectoryInfo("tests");

		private string GetRegion(string region, string fileName = "OverloadedMethods.cs")
		{
			var code = dir.GetFile(fileName).ContentAsUtf8();
			var extractor = new CommonSingleRegionExtractor(code);
			return extractor.GetRegion(new Label { Name = region });
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void ExtractAllFile()
		{
			Approvals.Verify(GetRegion("all"));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void ExtractOverlap1()
		{
			Approvals.Verify(GetRegion("methods_1_and_2"));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void ExtractOverlap2()
		{
			Approvals.Verify(GetRegion("methods_2_and_3"));
		}
	}
}