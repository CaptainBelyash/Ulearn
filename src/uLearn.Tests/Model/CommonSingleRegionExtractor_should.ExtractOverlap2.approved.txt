
public string Method(int a)
{
	return a.ToString("0");
}

//endregion methods_1_and_2

private string Method(string a)
{
	return a;
}

